scripts/check-components

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/ruby
+# coding: utf-8
+
+require 'find'
+
+COMMON_EVENTS = ['EventsReceived', 'EventsSent']
+SOURCE_EVENTS = ['BytesReceived'] + COMMON_EVENTS
+SINK_EVENTS = COMMON_EVENTS + ['BytesSent']
+# Transforms implicitly handle the common events in the topology, and
+# have no other emit requirements.
+TRANSFORM_EVENTS = []
+
+$error_count = 0
+
+def hash_array_add(hash, key, item)
+  arr = hash.fetch(key, Array::new)
+  arr.append(item)
+  hash[key] = arr
+end
+
+# Reduce a path to a component name, used to group all the emits for a
+# component that may be split over multiple files.
+def path_component(path)
+  parts = path.split('/')
+  parts.shift # Drop the leading 'src'
+  group = parts.shift
+  component = parts.shift
+
+  # Skip the top-level module files and utilities
+  if component == 'mod.rs' or component == 'util'
+    return nil
+  end
+
+  if component == 'gcp'
+    component = "gcp_#{parts.shift}"
+  end
+
+  if component.end_with? '.rs'
+    component = component[..-4]
+  end
+  return "#{group}/#{component}"
```

Comment:
nit: the last value doesn't need an explicit `return`. But it's fine to have it.

Reply:
Ah, I wondered if there was an implicit return.